flink-connectors/flink-connector-elasticsearch6/src/main/java/org/apache/flink/streaming/connectors/elasticsearch6/Elasticsearch6UpsertTableSink.java

```diff
@@ -159,12 +165,18 @@ protected ElasticsearchUpsertTableSinkBase copy(
 		Optional.ofNullable(sinkOptions.get(BULK_FLUSH_BACKOFF_DELAY))
 			.ifPresent(v -> builder.setBulkFlushBackoffDelay(Long.valueOf(v)));
 
-		builder.setRestClientFactory(
-			new DefaultRestClientFactory(
-				Optional.ofNullable(sinkOptions.get(REST_MAX_RETRY_TIMEOUT))
-					.map(Integer::valueOf)
-					.orElse(null),
-				sinkOptions.get(REST_PATH_PREFIX)));
+		String username = sinkOptions.get(USERNAME);
+		String password = sinkOptions.get(PASSWORD);
+		Integer maxRetryTimeout = Optional.ofNullable(sinkOptions.get(REST_MAX_RETRY_TIMEOUT))
+			.map(Integer::valueOf)
+			.orElse(null);
+		String pathPrefix = sinkOptions.get(REST_PATH_PREFIX);
+
+		if (Optional.ofNullable(username).isPresent() && Optional.ofNullable(password).isPresent()) {
```

Comment:
What is the benefit we get compared to `username != null`?

Reply:
@KarmaGYZ  I see other code use `Optional.ofNullable`, so i follow, the benefit is same to `username != null`